include/extractor/io.hpp

```diff
@@ -0,0 +1,62 @@
+#ifndef OSRM_EXTRACTOR_IO_HPP
+#define OSRM_EXTRACTOR_IO_HPP
+
+#include "extractor/segment_data_container.hpp"
+
+#include "storage/io.hpp"
+
+namespace osrm
+{
+namespace extractor
+{
+namespace io
+{
+
+template <>
+void read(const boost::filesystem::path &path, SegmentDataContainer &segment_data)
```

Comment:
Because `SegmentDataContainer = detail::SegmentDataContainer<UseSharedMemory=false>`. This could extended to allow the shared memory version, I'm not sure this makes sense though.